src/operations/delete.ts

```diff
@@ -25,7 +22,7 @@ export interface DeleteOptions extends CommandOperationOptions, WriteConcernOpti
 }
 
 /** @public */
-export interface DeleteResult {
+export interface DeleteResult extends Document {
```

Comment:
Why are we extending `Document`?

Reply:
When you do res.code, res.writeErrors, etc. in DeleteOneOperation and DeleteManyOperation, it errors saying that code and writeErrors do not exist on DeleteResult

Reply:
What happens if you change the definition to something like the following (note: I'm guessing on the types of writeErrors and code but I think this should fix the error that the properties don't exist on the delete result):

```typescript
export class DeleteOperation<DeleteResult> extends CommandOperation<DeleteResult & { code?: number, writeErrors?: Error[] } > {
```

Reply:
this still throws errors, specifically because there's a third property (n) with type number. when I include that property, the error is that "type "number | undefined" is not assignable to type "number""

Reply:
oh wait with the cleanup from below it was able to work